functorch/_src/aot_autograd.py

```diff
@@ -908,56 +960,36 @@ def functional_call(*args, **kwargs):
             )
         return out
 
-    def aot_function_simplified(
-        fn: Callable,
-        fw_compiler: Callable,
-        bw_compiler: Optional[Callable] = None,
-        partition_fn: Callable = default_partition,
-        decompositions: Optional[Dict] = None,
-        hasher_type=None,
-        static_argnums=None,
-    ) -> Callable:
-        assert static_argnums is None
-        if bw_compiler is None:
-            bw_compiler = fw_compiler
-        aot_config = AOTConfig(
-            fw_compiler=fw_compiler,
-            bw_compiler=bw_compiler,
-            partition_fn=partition_fn,
-            decompositions=decompositions,
-            num_params_buffers=params_len,
-        )
-
-        compiled_fn = None
+    assert static_argnums is None
+    if bw_compiler is None:
+        bw_compiler = fw_compiler
 
-        @wraps(fn)
-        def new_func(*args):
-            nonlocal compiled_fn
-            if compiled_fn is None:
-                compiled_fn = create_aot_dispatcher_function(
-                    fn,
-                    args,
-                    aot_config,
-                )
-            return compiled_fn(args)
+    aot_config = AOTConfig(
+        fw_compiler=fw_compiler,
+        bw_compiler=bw_compiler,
+        partition_fn=partition_fn,
+        decompositions=decompositions,
+        num_params_buffers=params_len,
+    )
 
-        return new_func
+    fake_flat_tensor_params.extend(inputs)
+    aot_dispatcher_function = _create_aot_dispatcher_function(functional_call, fake_flat_tensor_params, aot_config, fake_mode)
```

Comment:
This is a good refactor, I extracted it out into a standalone commit for ease of review: https://github.com/pytorch/pytorch/pull/89664